crates/relayer/src/chain.rs

```diff
@@ -6,60 +6,3 @@ pub mod handle;
 pub mod requests;
 pub mod runtime;
 pub mod tracking;
-
-use serde::{de::Error, Deserialize, Serialize};
-
-// NOTE(new): When adding a variant to `ChainType`, make sure to update
-//            the `Deserialize` implementation below and the tests.
-//            See the NOTE(new) comments below.
-
-#[derive(Copy, Clone, Debug, PartialEq, Eq, Serialize)]
-/// Types of chains the relayer can relay to and from
-pub enum ChainType {
-    /// Chains based on the Cosmos SDK
-    CosmosSdk,
-}
```

Comment:
This is gone entirely. Rather than having the config store an enum _of_ chain types, the config _is_ an enum, so that it can have different subtypes for different configs.